### PR TITLE
enhancement for bind points

### PR DESCRIPTION
**Pull request by yasin-cs-ko-ak**
This PR Contains enhancement for bind points aka SYS_BIND
- NETLINK Protocol added
- TCPv6 and TCP protocols added
- UNIX bindaddr added to sun_path

Here is few sample outputs:
```
Bind Points
+----------+---------------------+-----------+--------------+-------+------------------------------+
| PROTOCOL |       COMMAND       | BIND PORT | BIND ADDRESS | COUNT |      LAST UPDATED TIME       |
+----------+---------------------+-----------+--------------+-------+------------------------------+
| NETLINK  | /bin/ping           |           |              | 1     | Sat Dec 10 15:09:26 IST 2022 |
| TCP      | /bin/nc.traditional | 4444      | 0.0.0.0      | 1     | Sat Dec 10 15:10:29 IST 2022 |
+----------+---------------------+-----------+--------------+-------+------------------------------+
```
```
Bind Points
+----------+---------------------------+-----------+--------------+-------+------------------------------+
| PROTOCOL |          COMMAND          | BIND PORT | BIND ADDRESS | COUNT |      LAST UPDATED TIME       |
+----------+---------------------------+-----------+--------------+-------+------------------------------+
| NETLINK  | /usr/lib/apt/methods/http |           |              | 6     | Sat Dec 10 23:54:37 IST 2022 |
| TCP      | /usr/sbin/traceroute      | 0         | 0.0.0.0      | 14    | Sat Dec 10 23:55:07 IST 2022 |
| NETLINK  | /bin/ping                 |           |              | 1     | Sat Dec 10 23:56:41 IST 2022 |
+----------+---------------------------+-----------+--------------+-------+------------------------------+
```

Signed-off-by: yasin-cs-ko-ak <yasin@accuknox.com>

### CODE REVIEW

**Review by seswarrajan — CHANGES_REQUESTED** [review]

File: src/observability/summarizer.go [hunk 1]
```diff
@@ -60,6 +60,19 @@ func extractNetworkInfoFromSystemLog(netLog pb.Log) (string, string, string, str
 			if strings.Contains(locres, "sin_addr") {
 				bindAddress = strings.Split(locres, "=")[1]
 			}
+			if strings.Contains(locres, "AF_UNIX") && strings.Contains(locres, "sun_path") {
```
Comment (seswarrajan) [inline src/observability/summarizer.go]:
```suggestion
			if strings.Contains(netLog.Resource, "AF_UNIX") && strings.Contains(locres, "sun_path") {
```

**Review by seswarrajan — CHANGES_REQUESTED** [review]

File: src/observability/summarizer.go (hunk 1 above)
Comment (yasin-cs-ko-ak) [reply src/observability/summarizer.go]: This is currently not showing the sun_path in bindAddress.

**Comment by yasin-cs-ko-ak** [comment]
@seswarrajan @nyrahul Kindly review it.
and also to reflect this changes in karmor kindly merge this PR also: https://github.com/kubearmor/kubearmor-client/pull/219

**Review by seswarrajan — APPROVED** [review]
LGTM

File: src/observability/summarizer.go [hunk 2]
```diff
@@ -60,6 +75,18 @@ func extractNetworkInfoFromSystemLog(netLog pb.Log) (string, string, string, str
...
+				if protocol == "AF_INET6" {
+					protocol = strings.Replace(protocol, "AF_INET6", "TCPv6", 1)
+				} else if protocol == "AF_NETLINK" {
+					protocol = strings.Replace(protocol, "AF_NETLINK", "NETLINK", 1)
+				} else if protocol == "AF_INET" {
+					protocol = strings.Replace(protocol, "AF_INET", "TCP", 1)
+				} else if protocol == "AF_UNIX" {
+					protocol = strings.Replace(protocol, "AF_UNIX", "UNIX", 1)
+				}
+			}
```
Comment (nyrahul) [inline src/observability/summarizer.go]:
```suggestion
			if strings.Contains(locres, "sa_family") {
				protocol = strings.Split(locres, "=")[1]
			}
```

**Review by nyrahul — CHANGES_REQUESTED** [review]

File: src/observability/summarizer.go (hunk 2 above)
Comment (nyrahul) [reply src/observability/summarizer.go]: It is better to show protocol name as it is. Also there are more types of AF_* families out there which are not considered in this condition. If any of it shows up the value will be empty.

Comment (yasin-cs-ko-ak) [reply src/observability/summarizer.go]: Thank you for reviewing. Changes are done. Now table will look like this.
```

Bind Points
+------------+---------------------------+-----------+--------------+-------+------------------------------+
|  PROTOCOL  |          COMMAND          | BIND PORT | BIND ADDRESS | COUNT |      LAST UPDATED TIME       |
+------------+---------------------------+-----------+--------------+-------+------------------------------+
| AF_NETLINK | /usr/lib/apt/methods/http |           |              | 2     | Thu Dec 15 15:46:41 IST 2022 |
| AF_INET    | /bin/nc.traditional       | 4444      | 0.0.0.0      | 1     | Thu Dec 15 15:48:03 IST 2022 |
+------------+---------------------------+-----------+--------------+-------+------------------------------+
```

File: src/observability/summarizer.go [hunk 3]
```diff
@@ -51,7 +51,12 @@ func extractNetworkInfoFromSystemLog(netLog pb.Log) (string, string, string, str
 				}
 			}
 		}
-	} else if strings.Contains(netLog.Data, "SYS_BIND") {
+	}
+
+	if strings.Contains(netLog.Data, "SYS_BIND") {
+		bindAddress = ip
+	}
+	if strings.Contains(netLog.Data, "SYS_BIND") {
```
Comment (nyrahul) [inline src/observability/summarizer.go]:
```suggestion
    	if strings.Contains(netLog.Data, "SYS_BIND") {
	    	bindAddress = ip
	}

```
what is the point of using two different `if` conditions with the same rule?

**Review by nyrahul — CHANGES_REQUESTED** [review]

File: src/observability/summarizer.go (hunk 3 above)
Comment (yasin-cs-ko-ak) [reply src/observability/summarizer.go]: Changed.

**Review by nyrahul — APPROVED** [review]

**Pull request merged**